### ISSUE TITLE
%.16d processing does the zero padding inappropriately early

### ISSUE THREAD

**Issue by duffy-ocraven** [zeek/spicy#571]
```
module Issue_571;

import spicy;

public type Stream = unit {
    floats: oneFloat[];
};

const FLOAT_SIZE = 8;
type oneFloat = unit {
  data: bytes &size=FLOAT_SIZE;
  asDouble: real &type=spicy::RealType::IEEE754_Double &parse-from = self.data;
    on %done {
      print ("for the inf's and nan %%.16d prints %.16d" % (self.asDouble,));
      print ("for the inf's and nan %%.16g prints %.16g" % (self.asDouble,));
      print ("for the inf's and nan %%.16f prints %.16f" % (self.asDouble,));
    }
};
```
produces
```
$ printf "\377\370\000\000\000\000\000\000" | ../spicy/build/bin/spicy-driver Issue_571.spicy
for the inf's and nan %.16d prints 0000000000000nan
for the inf's and nan %.16g prints nan
for the inf's and nan %.16f prints nan

$ printf "\377\360\000\000\000\000\000\000" | ../spicy/build/bin/spicy-driver Issue_571.spicy
for the inf's and nan %.16d prints -000000000000inf
for the inf's and nan %.16g prints -inf
for the inf's and nan %.16f prints -inf

$ printf "\377\350\000\000\000\000\000\000" | ../spicy/build/bin/spicy-driver Issue_571.spicy
for the inf's and nan %.16d prints -1.348269851146737e+308
for the inf's and nan %.16g prints -1.348269851146737e+308
for the inf's and nan %.16f prints -134826985114673693079697889309176855021348273420672992955072560868299506854125722349531357991805652015840085409903545018244092326610812466869635572979605593283325920068649113957226664700934570589589812214063754326628613011756847161105434832905620427872512883013439723679960434453859787228626517247218168102912.0000000000000000
```

**Comment by bbannier** [comment]
This is an issue in the library we use for formatting. I created an upstream issue, https://github.com/c42f/tinyformat/issues/76.

**Comment by bbannier** [comment]
A first fix for this has been merged upstream, c42f/tinyformat#77. It looks like we still something along the lines of https://github.com/c42f/tinyformat/pull/78 to be able to consume that version though.

**Comment by bbannier** [comment]
This is now unblocked since the upstream issue was fixed by https://github.com/c42f/tinyformat/pull/80.
